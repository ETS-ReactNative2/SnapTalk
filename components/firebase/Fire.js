import FirebaseKeys from './Config';
import firebase from 'firebase';
require('firebase/firestore');

class Fire {

  constructor() {
    firebase.initializeApp(FirebaseKeys);
  }

  addPost = async ({ text, localUri, postKey }) => {
    //Uploading picture to database
    const remoteUri = await this.uploadPhotoAsync(localUri, 'photos/' + this.uid + '/' + Date.now());

    //Getting User from database
    const user = await firebase
      .firestore()
      .collection('users')
      .doc(this.uid)
      .get();

    //Pointer to the name field of the user in the database
    const fieldPathName = new firebase.firestore.FieldPath('name');

    //Getting User from database again (because cannot use the same variable)
    const userAgain = await firebase
      .firestore()
      .collection('users')
      .doc(this.uid)
      .get();

    //Pointer to the profile picture field of the user in the database
    const fieldPathProfilePicture = new firebase.firestore.FieldPath('profilePicture');

    //Creating a new post in the database
    return new Promise((res, rej) => {
      this.firestore
        .collection('posts')
        .add({
          text,
          uid: this.uid,
          postKey: '',
          timestamp: this.timestamp,
          image: remoteUri,
          username: user.get(fieldPathName),
          avatar: userAgain.get(fieldPathProfilePicture),
          listOfComments: [],
          nbOfComments: 0,
          listOfLikes: [],
        })
        .then(ref => {
          res(ref);
        })
        .catch(error => {
          rej(error);
        });
    });
  };

  updateUserLikedList = async postId => {
    //Getting post from the database
    let db = this.firestore.collection('posts').doc(postId);

    //Getting post from the database
    const post = await firebase
      .firestore()
      .collection('posts')
      .doc(postId)
      .get();

    //Creating pointer to the list Of likes of the post in the database
    const fieldPathListOfLikes = new firebase.firestore.FieldPath('listOfLikes');

    //Getting the list of likes of the post from the database
    const arrayOfLikes = await post.get(fieldPathListOfLikes);

    if (arrayOfLikes.includes(this.uid)) {
      db.update({
        listOfLikes: firebase.firestore.FieldValue.arrayRemove(this.uid),
      });
    }
    else {
      db.update({
        listOfLikes: firebase.firestore.FieldValue.arrayUnion(this.uid),
      });
    }
  };

  addPostKey = async postId => {
    //Getting post from database
    let dbPost = this.firestore.collection('posts').doc(postId);

    if (postId) {
      dbPost.update({
        postKey: postId,
      });
    }
  };

  updatePostList = async postId => {
    //Getting user from the database
    let db = this.firestore.collection('users').doc(this.uid);

    //Getting user from the database
    const user = await firebase
      .firestore()
      .collection('users')
      .doc(this.uid)
      .get();

    //Creating pointer to the list of posts field in the database
    const fieldPathListOfPosts = new firebase.firestore.FieldPath('listOfPosts');

    if (postId) {
      db.update({
        listOfPosts: firebase.firestore.FieldValue.arrayUnion(postId),
      });

      const nbOfPosts = (await user.get(fieldPathListOfPosts).length) + 1;

      db.set({ nbOfPosts: nbOfPosts }, { merge: true });
    }
  };

  addComment = async ({ text, postKey }) => {
    //Getting user from database
    const user = await firebase
      .firestore()
      .collection('users')
      .doc(this.uid)
      .get();

    //Creating a pointer to name field of the user in the database
    const fieldPathName = new firebase.firestore.FieldPath('name');

    //Getting User from database again (because cannot use the same variable)
    const userAgain = await firebase
      .firestore()
      .collection('users')
      .doc(this.uid)
      .get();

    //Creating a pointer to the profile picture field in the database
    const fieldPathProfilePicture = new firebase.firestore.FieldPath('profilePicture');

    return new Promise((res, rej) => {
      this.firestore
        .collection('comments')
        .add({
          comment: text,
          uid: this.uid,
          timestamp: this.timestamp,
          username: user.get(fieldPathName),
          postKey: postKey,
          commentKey: '',
          avatar: userAgain.get(fieldPathProfilePicture),
        })
        .then(ref => {
          res(ref);
        })
        .catch(error => {
          rej(error);
        });
    });
  };

  addCommentKey = async commentId => {
    //Getting comments from database
    let dbComment = this.firestore.collection('comments').doc(commentId);

    if (commentId) {
      dbComment.update({
        commentKey: commentId,
      });
    }
  };

  updateCommentList = async ({ commentId, postId }) => {
    //Getting user from database
    let dbUser = this.firestore.collection('users').doc(this.uid);

    //Getting post from database
    let dbPost = this.firestore.collection('posts').doc(postId);

    //Getting post again from database
    const post = await firebase
    .firestore()
    .collection('posts')
    .doc(postId)
    .get();

    //Getting user again from database
    const user = await firebase
      .firestore()
      .collection('users')
      .doc(this.uid)
      .get();

    //Creating pointer to the list of comments field in the database
    const fieldPathListOfComments = new firebase.firestore.FieldPath('listOfComments');

    if (commentId) {
      dbUser.update({
        listOfComments: firebase.firestore.FieldValue.arrayUnion(commentId),
      });

      const nbOfComments = (await user.get(fieldPathListOfComments).length) + 1;

      dbUser.set({ nbOfComments: nbOfComments }, { merge: true });
    }

    if (postId) {
      dbPost.update({
        listOfComments: firebase.firestore.FieldValue.arrayUnion(commentId)
      });

      const nbOfComments = (await post.get(fieldPathListOfComments).length) + 1;

      dbPost.set({ nbOfComments: nbOfComments }, { merge: true });
    }
  };

  modifyComment = async (commentKey, newComment) => {
    //Getting comments from database
    let dbComment = this.firestore.collection('comments').doc(commentKey);

    if (commentKey) {
      dbComment.update({
        comment: newComment,
      });
    }
  };

  deleteComment = async (commentId, postId) => {

    //Getting comment from database
    let dbComments = this.firestore.collection('comments').doc(commentId);

    //Getting user from database
    let dbUser = this.firestore.collection('users').doc(this.uid);

    //Getting post from database
    let dbPost = this.firestore.collection('posts').doc(postId);

    //Getting post again from database
    const post = await firebase
    .firestore()
    .collection('posts')
    .doc(postId)
    .get();

    //Getting user again from database
    const user = await firebase
      .firestore()
      .collection('users')
      .doc(this.uid)
      .get();

    //Creating pointer to the list of comments field in the database
    const fieldPathListOfComments = new firebase.firestore.FieldPath('listOfComments');

    if (commentId) {

      dbComments.delete();

      dbUser.update({
        listOfComments: firebase.firestore.FieldValue.arrayRemove(commentId)
      });

      const nbOfComments = (await user.get(fieldPathListOfComments).length) - 1;

      dbUser.set({ nbOfComments: nbOfComments }, { merge: true });
    }

    if (postId) {
      dbPost.update({
        listOfComments: firebase.firestore.FieldValue.arrayRemove(commentId),
      });

      const nbOfComments = (await post.get(fieldPathListOfComments).length) - 1;

      dbPost.set({ nbOfComments: nbOfComments }, { merge: true });
    }
  };



  createUser = async user => {
    let remoteAvatarUri = null;

    try {

      if (!user.name) {
        throw new Error('Username was not entered.');
      }

      await this.firestore
        .collection('users')
        .get()
        .then(snapshot => {
          snapshot.forEach(doc => {
            if (user.name == doc.data().name) {
              throw new Error('Username already taken.');
            }
          });
        });

      await firebase
        .auth()
        .createUserWithEmailAndPassword(user.email, user.password);

      let db = this.firestore.collection('users').doc(this.uid);

      db.set({
        name: user.name,
        email: user.email,
        listOfFollowers: [],
        listOfFollowing: [],
        listOfPosts: [],
        listOfComments: [],
        nbOfPosts: 0,
        nbOfComments: 0,
        profilePicture: remoteAvatarUri
      });

      if (user.avatar) {
        remoteAvatarUri = await this.uploadPhotoAsync(
          user.avatar,
          'avatar/' + this.uid,
        );
        db.set({ profilePicture: remoteAvatarUri }, { merge: true });
      }

    }
    catch (error) {
      alert('Error: ' + error.message);
    }
  };

  uploadPhotoAsync = async (uri, filename) => {
    //Uploading picture to firebase storage
    return new Promise(async (res, rej) => {
      const response = await fetch(uri);
      const file = await response.blob();

      let upload = firebase
        .storage()
        .ref(filename)
        .put(file);

      upload.on(
        'state_changed',
        snapshot => { },
        err => {
          rej(err);
        },
        async () => {
          const url = await upload.snapshot.ref.getDownloadURL();
          res(url);
        },
      );
    });
  };

  get firestore() {
    return firebase.firestore();
  }

  get uid() {
    return (firebase.auth().currentUser || {}).uid;
  }

  get timestamp() {
    return Date.now();
  }

}

Fire.shared = new Fire();
export default Fire;