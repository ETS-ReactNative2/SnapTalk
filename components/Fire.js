import FirebaseKeys from "./config";
import firebase from 'firebase';
require("firebase/firestore");

class Fire {

    constructor() {
        firebase.initializeApp(FirebaseKeys);
    }


    addPost = async ({ text, localUri, postKey }) => {
        const remoteUri = await this.uploadPhotoAsync(localUri, 'photos/' + this.uid + '/' + Date.now());

        const user = await firebase.firestore().collection("users").doc(this.uid).get();

        const fieldPathName = new firebase.firestore.FieldPath('name');

        const userAgain = await firebase.firestore().collection("users").doc(this.uid).get();

        const fieldPathProfilePicture = new firebase.firestore.FieldPath('profilePicture');

        return new Promise((res, rej) => {
            this.firestore.collection("posts").add({
                text,
                uid: this.uid,
                postKey: '',
                timestamp: this.timestamp,
                image: remoteUri,
                username: user.get(fieldPathName),
                avatar: userAgain.get(fieldPathProfilePicture),
                listOfComments: [],
                nbOfComments: 0,
                listOfLikes: [],
                nbOfLikes: 0,
            })
                .then(ref => {
                    res(ref);
                })
                .catch(error => {
                    rej(error);
                })
        })
    };

    updateUserLikedList = async (postId) => {
        let db = this.firestore.collection("posts").doc(postId);

        const post = await firebase.firestore().collection("posts").doc(postId).get();
        const fieldPathListOfLikes = new firebase.firestore.FieldPath('listOfLikes');
        const arrayOfLikes = await post.get(fieldPathListOfLikes)

        if (arrayOfLikes.includes(this.uid)) {
            db.update({
                listOfLikes: firebase.firestore.FieldValue.arrayRemove(this.uid),
            })
        }
        else {
            db.update({
                listOfLikes: firebase.firestore.FieldValue.arrayUnion(this.uid),
            })
        }
    }



    addPostKey = async (postId) => {

        let dbPost = this.firestore.collection("posts").doc(postId);

        if (postId) {
            dbPost.update({
                postKey: postId
            })
        }
    };

    updatePostList = async (postId) => {
        let db = this.firestore.collection("users").doc(this.uid);

        const user = await firebase.firestore().collection("users").doc(this.uid).get();

        const fieldPathListOfPosts = new firebase.firestore.FieldPath('listOfPosts');

        if (postId) {
            db.update({
                listOfPosts: firebase.firestore.FieldValue.arrayUnion(postId)
            })

            const nbOfPosts = await user.get(fieldPathListOfPosts).length + 1;

            db.set(
                { nbOfPosts: nbOfPosts }, { merge: true }
            )
        }
    }

    addComment = async ({ text, postKey }) => {
        const user = await firebase.firestore().collection("users").doc(this.uid).get();

        const fieldPathName = new firebase.firestore.FieldPath('name');

        const userAgain = await firebase.firestore().collection("users").doc(this.uid).get();

        const fieldPathProfilePicture = new firebase.firestore.FieldPath('profilePicture');

        return new Promise((res, rej) => {
            this.firestore.collection("comments").add({
                comment: text,
                uid: this.uid,
                timestamp: this.timestamp,
                username: user.get(fieldPathName),
                postKey: postKey,
                commentKey: '',
                avatar: userAgain.get(fieldPathProfilePicture)
            })
                .then(ref => {
                    res(ref);
                })
                .catch(error => {
                    rej(error);
                })
        })
    };

    addCommentKey = async (commentId) => {

        let dbComment = this.firestore.collection("comments").doc(commentId);

        if (commentId) {
            dbComment.update({
                commentKey: commentId
            })
        }
    };

    updateCommentList = async ({ commentId, postId }) => {
        let dbUser = this.firestore.collection("users").doc(this.uid);

        let dbPost = this.firestore.collection("posts").doc(postId);

        const user = await firebase.firestore().collection("users").doc(this.uid).get();

        const fieldPathListOfComments = new firebase.firestore.FieldPath('listOfComments');

        if (commentId) {
            dbUser.update({
                listOfComments: firebase.firestore.FieldValue.arrayUnion(commentId)
            })

            const nbOfComments = await user.get(fieldPathListOfComments).length + 1;

            dbUser.set(
                { nbOfComments: nbOfComments }, { merge: true }
            )
        }

        if (postId) {
            dbPost.update({
                listOfComments: firebase.firestore.FieldValue.arrayUnion(commentId)
            })

            const nbOfComments = await user.get(fieldPathListOfComments).length + 1;

            dbPost.set(
                { nbOfComments: nbOfComments }, { merge: true }
            )
        }
    }

    createUser = async user => {
        let remoteAvatarUri = null

        try {
            if (!user.name) {
                throw new Error("Username was not entered.")
            }

            await this.firestore
                .collection("users")
                .get()
                .then(snapshot => {
                    snapshot.forEach(doc => {
                        if (user.name == doc.data().name) {
                            throw new Error("Username already taken.")
                        }
                    })
                })

            await firebase.auth().createUserWithEmailAndPassword(user.email, user.password);

            let db = this.firestore.collection("users").doc(this.uid);

            db.set({
                name: user.name,
                email: user.email,
                listOfFollowers: [],
                listOfFollowing: [],
                listOfPosts: [],
                listOfComments: [],
                nbOfPosts: 0,
                nbOfComments: 0,
                profilePicture: remoteAvatarUri
            });

            if (user.avatar) {
                remoteAvatarUri = await this.uploadPhotoAsync(user.avatar, 'avatar/' + this.uid);
                db.set({ profilePicture: remoteAvatarUri }, { merge: true })
            }

        } catch (error) {
            alert("Error: " + error.message);
        }
    }

    uploadPhotoAsync = async (uri, filename) => {

        return new Promise(async (res, rej) => {
            const response = await fetch(uri);
            const file = await response.blob();

            let upload = firebase
                .storage()
                .ref(filename)
                .put(file);

            upload.on("state_changed", snapshot => { }, err => {
                rej(err);
            },
                async () => {
                    const url = await upload.snapshot.ref.getDownloadURL();
                    res(url);
                }
            );
        });
    };


    get firestore() {
        return firebase.firestore();
    }

    get uid() {
        return (firebase.auth().currentUser || {}).uid;
    }

    get timestamp() {
        return Date.now();
    }
}


Fire.shared = new Fire();
export default Fire;