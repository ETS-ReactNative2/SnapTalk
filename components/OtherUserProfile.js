import React, { Component } from 'react';
import {StyleSheet,View,Text, Image, Button, TouchableHighlight, Modal, Alert} from 'react-native';
import Fire from './Fire';


export default class ModalExample extends Component {
  state = {
    modalVisible: false,
  };

  setModalVisible(visible) {
    this.setState({modalVisible: visible});
  }

  render() {
    return (
      <View style={{marginTop: 22}}>
        <Modal
          animationType="slide"
          transparent={false}
          visible={this.state.modalVisible}
          onRequestClose={() => {
            Alert.alert('Modal has been close');
          }}>
          <View style={{marginTop: 22}}>
            <View>
              <Text>{this.props.username}</Text>

              <TouchableHighlight
                onPress={() => {
                  this.setModalVisible(!this.state.modalVisible);
                }}>
                <Text>Hide Modal</Text>
              </TouchableHighlight>
            </View>
          </View>
        </Modal>

        <TouchableHighlight
          onPress={() => {
            this.setModalVisible(true);
          }}>
          <Text style={styles.modalProfile}>{this.props.username}</Text>
        </TouchableHighlight>
      </View>
    );
  }
}


const styles = StyleSheet.create({

  modalProfile:{
    paddingBottom:2

  }

});