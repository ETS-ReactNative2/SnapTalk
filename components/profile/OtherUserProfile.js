import React, {Component} from 'react';
import { StyleSheet, View, Text, Image, TouchableHighlight, Modal, ImageBackground } from 'react-native';
import Fire from '../firebase/Fire';
import FollowButton from '../profile/FollowButton';
import LogoutButton from '../profile/LogoutButton';
import Icon from 'react-native-vector-icons/Ionicons';

const firebase = require('firebase');
require('firebase/firestore');

export default class ModalExample extends Component {

  state = {
    modalVisible: false,
    user: {},
    nbOfFollowers: 0,
    nbOfFollowing: 0,
    nbOfPosts: 0,
    currentUserId: null,
    isNotSameUser: true
  };

  setModalVisible(visible) {
    this.setState({modalVisible: visible});
  }

  componentDidMount() {
    this.getUserId();
  }

  getListSize = async () => {
    const user = await firebase
      .firestore()
      .collection('users')
      .doc(this.props.postUserId)
      .get();

    const listOfPosts = new firebase.firestore.FieldPath('listOfPosts');

    this.setState({nbOfPosts: await user.get(listOfPosts).length});

    const listOfFollowers = new firebase.firestore.FieldPath('listOfFollowers');

    this.setState({nbOfFollowers: await user.get(listOfFollowers).length});

    const listOfFollowing = new firebase.firestore.FieldPath('listOfFollowing');

    this.setState({nbOfFollowing: await user.get(listOfFollowing).length});
  };

  getUserId = async () => {
    this.setState({isLoadingList: true});
    this.unsubscribe = await Fire.shared.firestore
      .collection('users')
      .get()
      .then(snapshot => {
        snapshot.forEach(doc => {
          if (doc.data().name == this.props.username) {
            this.setState({user: doc.data()});
            this.getListSize();
            var current = firebase.auth().currentUser;
            this.setState({currentUserId: current.uid});
            if (current.uid == doc.id) {
              this.setState({isNotSameUser: false});
            }
          }
        });
      });
  };

  render() {
    return (
      <View>
        <Modal
          animationType="slide"
          transparent={false}
          visible={this.state.modalVisible}
          onRequestClose={() => {
            this.setModalVisible(!this.state.modalVisible);
          }}>
          <View style={styles.modalContainer}>
            <View style={styles.topBar}>
              <TouchableHighlight
                onPress={() => {
                  this.setModalVisible(!this.state.modalVisible);
                }}>
                <Icon
                  name="ios-arrow-back"
                  size={24}
                  style={styles.backButton}
                />
              </TouchableHighlight>
              {!this.state.isNotSameUser && (
                <LogoutButton style={styles.logout} />
              )}
            </View>
            <View>
              <View styles={styles.container}>
                <View style={styles.header}>
                  <ImageBackground
                    source={require('../../assets/Default-profile-bg.jpg')}
                    style={styles.backgroundImage}>
                    <View style={styles.avatarContainer}>
                      <Image
                        style={styles.avatar}
                        source={
                          this.state.user.profilePicture
                            ? {uri: this.state.user.profilePicture}
                            : require('../../assets/tempAvatar.jpg')
                        }></Image>
                    </View>
                    <Text style={styles.name}> {this.state.user.name} </Text>
                  </ImageBackground>
                </View>
                <View style={styles.followStyle}>
                  {this.state.isNotSameUser && (
                    <FollowButton
                      loggedUserUID={this.state.currentUserId}
                      userToFollow={this.state.user.name}
                    />
                  )}
                </View>
                <View style={styles.info}>
                  <View style={styles.state}>
                    <Text style={styles.amount}> {this.state.nbOfPosts} </Text>
                    <Text style={styles.title}> Posts </Text>
                  </View>
                  <View
                    style={[
                      styles.state,
                      {
                        borderColor: '#DFD8C8',
                        borderLeftWidth: 1,
                        borderRightWidth: 1,
                      },
                    ]}>
                    <Text style={styles.amount}>
                      {' '}
                      {this.state.nbOfFollowers}{' '}
                    </Text>
                    <Text style={styles.title}> Followers </Text>
                  </View>
                  <View style={styles.state}>
                    <Text style={styles.amount}>
                      {' '}
                      {this.state.nbOfFollowing}{' '}
                    </Text>
                    <Text style={styles.title}> Following </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </Modal>
        <TouchableHighlight
          onPress={() => {
            this.setModalVisible(true);
          }}>
          <Text style={styles.modalProfile}>{this.props.username}</Text>
        </TouchableHighlight>
      </View>
    );
  }
  
}

const styles = StyleSheet.create({
  modalProfile: {
    paddingBottom: 2,
  },
  header:
  {
    paddingBottom: 10
  },
  backButton:
  {
    paddingVertical: 3
  },
  modalContainer:
  {
    backgroundColor: '#EFECF4'
  },
  container: {
    flex: 1,
  },
  name: {
    marginTop: 24,
    fontSize: 16,
    fontWeight: 'bold',
    paddingBottom: 10,
  },
  info: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    margin: 32,
    backgroundColor: '#FFF',
    borderRadius: 5,
    padding: 8,
    flexDirection: 'row',
    marginVertical: 8,
  },
  state: {
    alignItems: 'center',
    flex: 1,
  },
  amount: {
    fontSize: 18,
    color: '#52575D',
    fontFamily: 'HelveticaNeue',
  },
  title: {
    fontSize: 12,
    fontWeight: 'bold',
    marginTop: 4,
    color: '#AEB5BC',
    textTransform: 'uppercase',
    fontWeight: '500',
  },
  avatarContainer: {
    shadowColor: '#151734',
    shadowRadius: 30,
    shadowOpacity: 0.4,
    paddingTop: 10,
  },
  avatar: {
    width: 150,
    height: 150,
    borderRadius: 75,
    borderWidth: 5,
    borderColor: '#6495ED',
  },
  followStyle: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  logout: {
    alignSelf: 'flex-end',
  },
  backgroundImage: {
    alignItems: 'center',
    borderTopWidth: 1,
    borderColor: '#52575D',
  },
  topBar: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingHorizontal: 15,
    paddingVertical: 5,
  },
});
