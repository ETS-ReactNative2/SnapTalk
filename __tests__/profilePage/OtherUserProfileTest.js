import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import OtherUserProfile from '../../components/profile/OtherUserProfile'
import Fire from '../../components/firebase/Fire'

describe('OtherUserProfile Test', () => {
    let otherUserProfileSnapshot = renderer.create(<OtherUserProfile />).toJSON();

    test('OtherUserProfile Render Test', () => {
        expect(otherUserProfileSnapshot).toMatchSnapshot();
    })
})