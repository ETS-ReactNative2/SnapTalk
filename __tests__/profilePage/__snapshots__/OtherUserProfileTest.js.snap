// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OtherUserProfile Test OtherUserProfile Render Test 1`] = `
<View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={false}
  >
    <View
      style={
        Object {
          "backgroundColor": "#EFECF4",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 15,
            "paddingVertical": 5,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 24,
                },
                Object {
                  "paddingVertical": 3,
                },
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            Ôèè
          </Text>
        </View>
      </View>
      <View>
        <View
          styles={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "paddingBottom": 10,
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#52575D",
                  "borderTopWidth": 1,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../assets/Default-profile-bg.jpg",
                  }
                }
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "height": undefined,
                      "width": undefined,
                    },
                    undefined,
                  ]
                }
              />
              <View
                style={
                  Object {
                    "paddingTop": 10,
                    "shadowColor": "#151734",
                    "shadowOpacity": 0.4,
                    "shadowRadius": 30,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../assets/tempAvatar.jpg",
                    }
                  }
                  style={
                    Object {
                      "borderColor": "#6495ED",
                      "borderRadius": 75,
                      "borderWidth": 5,
                      "height": 150,
                      "width": 150,
                    }
                  }
                />
              </View>
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginTop": 24,
                    "paddingBottom": 10,
                  }
                }
              >
                 
                 
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "white",
                  "borderColor": "black",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "height": 20,
                  "width": 70,
                }
              }
            >
              <Text
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                Follow
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#FFF",
                "borderRadius": 5,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "margin": 32,
                "marginVertical": 8,
                "padding": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#52575D",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 18,
                  }
                }
              >
                 
                0
                 
              </Text>
              <Text
                style={
                  Object {
                    "color": "#AEB5BC",
                    "fontSize": 12,
                    "fontWeight": "500",
                    "marginTop": 4,
                    "textTransform": "uppercase",
                  }
                }
              >
                 Posts 
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                  },
                  Object {
                    "borderColor": "#DFD8C8",
                    "borderLeftWidth": 1,
                    "borderRightWidth": 1,
                  },
                ]
              }
            >
              <Text
                style={
                  Object {
                    "color": "#52575D",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 18,
                  }
                }
              >
                 
                0
                 
              </Text>
              <Text
                style={
                  Object {
                    "color": "#AEB5BC",
                    "fontSize": 12,
                    "fontWeight": "500",
                    "marginTop": 4,
                    "textTransform": "uppercase",
                  }
                }
              >
                 Followers 
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#52575D",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 18,
                  }
                }
              >
                 
                0
                 
              </Text>
              <Text
                style={
                  Object {
                    "color": "#AEB5BC",
                    "fontSize": 12,
                    "fontWeight": "500",
                    "marginTop": 4,
                    "textTransform": "uppercase",
                  }
                }
              >
                 Following 
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <Text
      style={
        Object {
          "paddingBottom": 2,
        }
      }
    />
  </View>
</View>
`;
